nautobot_design_builder/design.py

```diff
@@ -403,12 +427,16 @@ def _load_instance(self):
                 self.attributes.update(query_filter)
         elif self.action != "create":
             raise errors.DesignImplementationError(f"Unknown database action {self.action}", self.model_class)
+
         self._initial_state = {}
         if not self.instance:
             self.created = True
-        self.instance = self.model_class()
+        try:
+            self.instance = self.model_class(**self._kwargs)
+        except TypeError as ex:
+            raise errors.DesignImplementationError(str(ex), self.model_class)
 
-    def _update_fields(self):  # pylint: disable=too-many-branches
+    def _update_fields(self, output_dict):  # pylint: disable=too-many-branches
```

Comment:
I really don't like the use of `output_dict` here. We're passing it around all over the place but really only updating it in one field type. The point of the `ModelInstance` class is to carry state that design builder needs to function. We should be leveraging `ModelInstance` instead of `output_dict`.